### PR TITLE
Correct menus

### PR DESCRIPTION
**Pull request by fcollonval**
Correct the menu example
Moves it into `main-menu`
Split it to create `commands` and `command-palette` example

Xref: #14

### CODE REVIEW

File: command-palette/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Command Palette: Extending the main app
+
+One major concept of the phosphorjs library on which JupyterLab is built is
```
Comment (jtpio) [inline command-palette/README.md]: Should we start using [Lumino](https://github.com/jupyterlab/lumino) directly, or wait a little bit longer?

File: command-palette/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+the notion of _Commands_ as explained in the [commands example](../commands/README.md).
+One of the possibility offered to the user to trigger that command is to call it from
```
Comment (jtpio) [inline command-palette/README.md]: Or?

```suggestion
Commands can be used from the command palette.
```

File: command-palette/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+The `ICommandPalette`
+([documentation](https://JupyterLab.github.io/JupyterLab/interfaces/_apputils_src_commandpalette_.icommandpalette.html))
+is passed to the `activate` function as an argument (variable `palette`) in
+addition to the JupyterLab application (variable `app`). We request that dependency
+with the property `requires: [ICommandPalette],`. It lists the additional arguments
+we want to inject into the `activate` function in the `JupyterFontEndPlugin`.
+
+`ICommandPalette` provides the method `addItem` that links a palette entry to a command in the command registry. It requires two arguments: the unique command id and the command
+category (that can be either an existing or a new one). And optionnaly, you can specify
```
Comment (jtpio) [inline command-palette/README.md]: Maybe useful to repeat "category" here:

```suggestion
category (that can be either an existing category or a new one). And optionally, you can specify
```

File: command-palette/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+the arguments that will be passed to the command when executed from the palette.
+
+When running JupyterLab with this extension active, the following message should
```
Comment (jtpio) [inline command-palette/README.md]: Do we need "active"?

```suggestion
When running JupyterLab with this extension, the following message should
```

File: commands/README.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+// src/index.ts#L27-L31
+
+commands.execute(command, { origin: 'from init' }).catch(reason => {
+  console.error(
+    `An error occurs during the execution of tutorial:command.\n${reason}`
+  );
+});
+```
+
+When running JupyterLab with this extension active, the following message should
```
Comment (jtpio) [inline commands/README.md]: Same as above: is the word "active" needed?

File: commands/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      caption: 'Execute tutorial:command',
+      execute: (args: any) => {
+        console.log(`tutorial:command has been called ${args['origin']}.`);
+      }
+    });
+
+    // Call the command execution
+    commands.execute(command, { origin: 'from init' }).catch(reason => {
+      console.error(
+        `An error occurs during the execution of tutorial:command.\n${reason}`
```
Comment (jtpio) [inline commands/src/index.ts]: Should we use the past tense instead?

```suggestion
        `An error occurred during the execution of tutorial:command.\n${reason}`
```

File: main-menu/README.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
+# Menus: Extending the main app
+
+One major concept of the phosphorjs library on which JupyterLab is built is
+the notion of _Commands_ as explained in the [commands example](../commands/README.md).
+One of the possibility offered to the user to trigger that command is to call it from
```
Comment (jtpio) [inline main-menu/README.md]:
```suggestion
One of the possibilities offered to the user to trigger that command is to call it from
```

File: main-menu/README.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+a menu item.
+
+Adding new menu items works in a similar way to the [command palette](../../command-palette/README.md).
```
Comment (jtpio) [inline main-menu/README.md]:
```suggestion
Adding new menu item works in a similar way to the [command palette](../../command-palette/README.md).
```

**Comment by jtpio** [comment]
That's great, thanks @fcollonval!

I left a couple of suggestions while reading through it, let me know what you think.

File: command-palette/README.md (hunk 1 above)
Comment (fcollonval) [reply command-palette/README.md]: I think you should stick to JLab 1.2 for now (and phosphorjs). And make the switch for JLab 2.0 release. What do you think? @echarles

File: command-palette/README.md (hunk 2 above)
Comment (fcollonval) [reply command-palette/README.md]: Thx this is much simpler.

**Comment by fcollonval** [comment]
Thanks Jeremy for the correction. I let this open in wait for Eric opinion on Lumino.

File: command-palette/README.md (hunk 1 above)
Comment (jtpio) [reply command-palette/README.md]: Sounds good, at least wait for the switch to happen in JLab.

**Comment by jtpio** [comment]
Alright!

**Pull request merged**